dali/ft/daftprogress.cpp

```diff
@@ -88,9 +90,12 @@ void DaftProgress::onProgress(unsigned __int64 sizeDone, unsigned __int64 totalS
         unsigned secsLeft = (unsigned)(timeLeft * cycleToNanoScale /1000000000);
         char temp[20];
         formatTime(temp, secsLeft);
-        displayProgress((unsigned)(sizeDone*100/totalSize), secsLeft, temp, 
-                        sizeDone/scale,totalSize/scale,scaleUnit, 
-                        (unsigned)(msGone ? (sizeDone-startSize)/msGone : 0),
+
+        unsigned percentDone = (unsigned)(totalSize ? (sizeDone*100/totalSize) : 0);
+
+        displayProgress(percentDone, secsLeft, temp,
+                        sizeDone/scale, totalSize/scale, scaleUnit,
+                        (unsigned)((sizeDone-startSize)/msGone),
```

Comment:
previously:
>(unsigned)(msGone ? (sizeDone-startSize)/msGone : 0)

this displayProgress param = 'kbPerSecondRate'.
It looks like this code was using the fact that divisor was milliseconds, to in effect make the divisor x 1000 so it approximated k per sec.

I think it should really be something like
```
kbPerSecond = (sizeDone-startSize) / 1024;
if (msGone) // if took no time, leave as max total kb.
    kbPerSecond /= (msGone / 1000);
```

Reply:
>displaySummary(temp, (unsigned)((totalSize - startSize)/msGone));

.. and the same based on totalSize on line 104